### DISCUSSION TITLE
How to hold on to toggle a layer and also execute a button press at the same time?

### DISCUSSION THREAD

**Discussion by txtyash** [kmonad/kmonad#396] [Q&A] [answered]
```
mod  (tap-macro lmet (layer-toggle qwerty))

```
I want to hold down on a key and want it to toggle a layer as well as hold down lmet key at the same time. 

Suppose map it to the super key then when I hold down the super key and as long as I've held it down, I want it to act like I have held down the lmet key and  I also want it to toggle a layer. How do I do it? The above binding doesn't work as expected.

**Comment by txtyash** [comment, marked as answer]
Make another layer full of bindings such as M-a M-s M-d,etc.  

So the binding would then be:

`mod (tap-hold 153 (layer-toggle qwerty) lmet)`

**Reply by MuhammedZakir** [reply]
In questions/posts where you figure the answer yourself, it is better to post your solution and mark that as answer. This way it will help others with similiar questions. :-)

**Reply by txtyash** [reply]
> In questions/posts where you figure the answer yourself, it is better to post your solution and mark that as answer. This way it will help others with similiar questions. :-)

oh okay
